torch/csrc/jit/codegen/cuda/manager.cpp

```diff
@@ -209,14 +210,32 @@ void runCudaFusionGroup(const Node* fusion_node, Stack& stack) {
   FUSER_PERF_SCOPE("nvFuser::Manager::runCudaFusionGroup");
 
   // Fallback to use if anything goes wrong
-  auto take_fallback = [&]() {
+  auto take_fallback = [&](Stack& stack) {
     // copying graph here since we are eliminating shape information;
     auto copied_graph = fusion_node->g(attr::Subgraph)->copy();
     EraseShapeInformation(copied_graph);
     enableAliasCopyNodes(copied_graph, copied_graph->block());
     InterpreterState{Code(copied_graph, "fallback_cuda_fuser")}.run(stack);
   };
 
+  c10::optional<Stack> stack_copy;
+  auto compare_callback = getCudaFuserComparisonCallback();
+  if (compare_callback.run_fallback) {
+    // make a copy of the stack
+    int64_t inputs_size =
+        static_cast<int64_t>(fusion_node->g(attr::Subgraph)->inputs().size());
+    TORCH_INTERNAL_ASSERT(stack.size() >= inputs_size);
+    stack_copy = Stack();
+    stack_copy->insert(
+        stack_copy->end(), stack.begin(), stack.end() - inputs_size);
```

Comment:
Should we assert on `!stack_copy.empty()`? Since we later (L297) only use to determine whether we re-run fallback.

Reply:
stack_copy is a c10::optional<> so conversion to bool will return true as long as it has a value - I can change the later if statement to `if (stack_copy.has_value())` for clarity

Reply:
sorry I missed the `optional` part... code looks fine as-is.